src/storage/core.rs

```diff
@@ -765,19 +785,36 @@ impl<K: Key> Storage<K> {
         let key = key.as_ref();
         trace!("[{:?}] check in blobs bloom filter", &key.to_vec());
         let inner = self.inner.safe.read().await;
-        let in_active = inner
-            .active_blob
-            .as_ref()
-            .map_or(false, |active_blob| active_blob.check_filters(key.as_ref()));
+        let in_active = if let Some(active_blob) = inner.active_blob.as_ref() {
+            active_blob.check_filters(key.as_ref()).await
+        } else {
+            false
+        };
         let in_closed = inner
             .blobs
             .read()
             .await
             .iter()
-            .any(|blob| blob.check_filters(key.as_ref()) == true);
+            .map(|blob| blob.check_filters(key.as_ref()))
+            .collect::<FuturesUnordered<_>>()
+            .any(|value| value)
+            .await;
```

Comment:
For performance reasons it is better to split this check into two sequential:
1. Check only those filters that are in memory. This is a quick check, and if it returns `true`, we don't need to perfrom checks on disk
2. If the previous step returned false, then run checks on disk

Reply:
Done